### PR TITLE
New Channels Layout in Subscriptions Tab and Option to change App startup page

### LINKED ISSUE #193: Update subscription tab.
### Describe your suggested feature

Subscribed channels icons are displayed in single row. Hard to see those items scrolling horizontally. Make it collapse/expand by clicking arrow on right which will display 3 column grid when expanded.

Also a unread notification dot can be shown over icons to indicate channel uploaded new unwatched video (this available in YouTube).

Remove home tab. This tab showing completely unrelated videos (same experience when looking at ads). Doubtful if people even click any. And set subscription tab as default as this showing actual videos from subscribed channels.

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by Bnyro**
closes #193 and #124 + #23

### CODE REVIEW

File: app/src/main/java/com/github/libretube/Subscriptions.kt [hunk 1]
```diff
@@ -66,6 +66,12 @@ class Subscriptions : Fragment() {
                 fetchFeed(feedRecView, progressBar, view)
             }
 
+            var toggleSubs = view.findViewById<RelativeLayout>(R.id.toggle_subs)
+            toggleSubs.setOnClickListener {
+                channelRecView.visibility = if (channelRecView.visibility == View.GONE) View.VISIBLE else View.GONE
+                feedRecView.visibility = if (feedRecView.visibility == View.GONE) View.VISIBLE else View.GONE
+            }
```
Comment (FireMasterK) [inline app/src/main/java/com/github/libretube/Subscriptions.kt:73]: Maybe we can lazy load the subscriptions api request here?

**Review by FireMasterK — COMMENTED** [review]
I have to note that UI performance when opening and closing the subscriptions feels slow, I haven't had a look at why this could be yet.

File: app/src/main/java/com/github/libretube/Subscriptions.kt (hunk 1 above)
Comment (Bnyro) [reply app/src/main/java/com/github/libretube/Subscriptions.kt]: Changed it. But the performance shouldn't have changed with my pr, idk

File: app/src/main/java/com/github/libretube/Subscriptions.kt [hunk 2]
```diff
@@ -67,9 +66,18 @@ class Subscriptions : Fragment() {
             }
 
             var toggleSubs = view.findViewById<RelativeLayout>(R.id.toggle_subs)
+            toggleSubs.visibility = View.VISIBLE
             toggleSubs.setOnClickListener {
-                channelRecView.visibility = if (channelRecView.visibility == View.GONE) View.VISIBLE else View.GONE
-                feedRecView.visibility = if (feedRecView.visibility == View.GONE) View.VISIBLE else View.GONE
+                if (!channelRecView.isVisible) {
+                    channelRecView?.layoutManager = GridLayoutManager(context, 4)
+                    fetchChannels(channelRecView)
```
[the remark this answers is not in the document]
Comment (Bnyro) [reply app/src/main/java/com/github/libretube/Subscriptions.kt:73]: That's right lol. Idk what I did there xD

**Comment by matkoniecz** [comment]
Where and how this setting can be accessed? I want to get rid of trends on startup but I cannot figure out where this setting is present.

**Pull request merged**